aidial_adapter_openai/utils/tokens.py

```diff
@@ -43,62 +44,23 @@ def calculate_tokens_per_message(self, message: dict) -> int:
 
         tokens = tokens_per_message
         for key, value in message.items():
-            if isinstance(value, str):
-                tokens += self.calculate_tokens(value)
             if key == "name":
                 tokens += tokens_per_name
-
+            elif isinstance(value, str):
+                tokens += self.calculate_tokens(value)
+            if key == "content" and isinstance(value, list):
+                for submessage in value:
+                    if submessage["type"] == "text":
+                        tokens += self.calculate_tokens(submessage["text"])
         return tokens
 
+    def calculate_overall_prompt_tokens(self, message_tokens: int):
```

Comment:
`overall` here actually means **the whole request**. So let's spell it out explicitly: `calculate_request_prompt_tokens`.

Also `message_tokens` is rather `messages_tokens`, because it supposed to be a sum of tokens for **all messages** in the request.

Reply:
Fixed